git-config/src/values.rs

```diff
@@ -300,24 +303,38 @@ impl Path<'_> {
             return Err(());
         }
 
-        if val.starts_with(b"~/") {
-            let mut home = env::var("HOME").map_err(|_| ())?;
-            if env::consts::OS == "windows" {
-                home = home.replace("\\", "/");
-            }
-            let mut expanded = vec![];
-            expanded.extend(home.as_bytes().into_iter());
-            if !home.ends_with("/") {
-                expanded.push(b'/');
-            }
-            expanded.extend(&val[2..]);
+        // TODO handle %(prefix) case
 
-            return Ok(Path {
+        const SLASH: u8 = b'/';
+
+        if val.starts_with(b"~/") {
```

Comment:
There is a lot of error cases and I think it's time for upgrading to `quickerror`. This will also do away with the manual error mapping in many if not all cases.